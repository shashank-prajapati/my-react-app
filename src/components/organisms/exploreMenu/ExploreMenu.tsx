import { Box, Divider, Grid, MenuList, Paper, Typography } from "@mui/material";
import React from "react";
import ExploreMenuButton from "../../molecules/exploreMenuButton/ExploreMenuButton";
import { ReactComponent as EntrepreneurshipIcon } from "../../../media/Vector.svg";
import { ReactComponent as CareerSuccess } from "../../../media/CareerSuccess.svg";
import { ReactComponent as CommunicationSkills } from "../../../media/Communication Skills.svg";
import { ReactComponent as CorporateCulture } from "../../../media/Corporate Culture.svg";
import { ReactComponent as Economics } from "../../../media/Economics.svg";
import { ReactComponent as Education } from "../../../media/Education.svg";
import { ReactComponent as HealthNutrition } from "../../../media/Health Nutrition.svg";
import { ReactComponent as History } from "../../../media/History.svg";
import { ReactComponent as MarketingSales } from "../../../media/Marketing Sales.svg";
import { ReactComponent as MoneyInvestments } from "../../../media/Money Investments.svg";
import { ReactComponent as Motivation } from "../../../media/Motivation Inspiration.svg";
import { ReactComponent as Nature } from "../../../media/Nature Environment.svg";
import { ReactComponent as PersonalDevelopment } from "../../../media/Personal Development.svg";
import { ReactComponent as Politics } from "../../../media/Politics.svg";
import { ReactComponent as Productivity } from "../../../media/Productivity.svg";
import { ReactComponent as Relationship } from "../../../media/SexRelationship.svg";
import { ReactComponent as Science } from "../../../media/Science.svg";
import { ReactComponent as Psycology } from "../../../media/Psychology.svg";

const menuData = [
  {
    id: 1,
    iconComponent: EntrepreneurshipIcon,
    buttonText: "Entrepreneurship",
    path: "entrepreneurship",
  },
  {
    id: 2,
    iconComponent: Politics,
    buttonText: "Politics",
    path: "politics",
  },
  {
    id: 3,
    iconComponent: MarketingSales,
    buttonText: "Marketing & Sales",
    path: "marketing-sales",
  },
  {
    id: 4,
    iconComponent: Science,
    buttonText: "Science",
    path: "science",
  },
  {
    id: 5,
    iconComponent: HealthNutrition,
    buttonText: "Health & Nutrition",
    path: "health-nutrition",
  },
  {
    id: 6,
    iconComponent: PersonalDevelopment,
    buttonText: "Personal Development",
    path: "personal-development",
  },
  {
    id: 7,
    iconComponent: Economics,
    buttonText: "Economics",
    path: "economics",
  },
  {
    id: 8,
    iconComponent: History,
    buttonText: "History",
    path: "history",
  },
  {
    id: 9,
    iconComponent: CommunicationSkills,
    buttonText: "Communication Skills",
    path: "communication-skills",
  },
  {
    id: 10,
    iconComponent: CorporateCulture,
    buttonText: "CorporateCulture",
    path: "corporate-culture",
  },
  {
    id: 11,
    iconComponent: Motivation,
    buttonText: "Motivation & Inspiration",
    path: "motivation-inspiration",
  },
  {
    id: 12,
    iconComponent: MoneyInvestments,
    buttonText: "Money & Investments",
    path: "money-investments",
  },
  {
    id: 13,
    iconComponent: Psycology,
    buttonText: "Psychology",
    path: "psychology",
  },
  {
    id: 14,
    iconComponent: Productivity,
    buttonText: "Productivity",
    path: "productivity",
  },
  {
    id: 15,
    iconComponent: Relationship,
    buttonText: "Sex & Relationship",
    path: "sex-relationship",
  },
  {
    id: 16,
    iconComponent: Nature,
    buttonText: "Nature & Environment",
    path: "nature-environment",
  },
  {
    id: 17,
    iconComponent: CareerSuccess,
    buttonText: "Career & Success",
    path: "career-success",
  },
  {
    id: 18,
    iconComponent: Education,
    buttonText: "Education",
    path: "education",
  },
];

type Props = {
  onClick?: any;
};

function ExploreMenu(props: Props) {
  const { onClick } = props;
  const menuList = menuData.map((menuItem) => (
    <Grid item xs={4}>
      <ExploreMenuButton
        key={menuItem.id}
        iconComponent={menuItem.iconComponent}
        buttonText={menuItem.buttonText}
        onClick={() => onClick(menuItem.path)}
      />
    </Grid>
  ));

  return (
    <Paper
      sx={{
        bgcolor: "footerBgColor.main",
        px: "250px",
        py: "30px",
        mx: 0,
      }}
      elevation={0}
    >
      <Box
        sx={{
          display: "flex",
          justifyContent: "space-between",
        }}
      >
        <Typography variant="subtitle3" color="secondary.main">
          Explore by category
        </Typography>
        <Typography variant="subtitle3" color="extraTextColor.main">
          See recently added titles
        </Typography>
        <Typography variant="subtitle3" color="extraTextColor.main">
          See popular titles
        </Typography>
      </Box>
      <Divider sx={{ my: 2 }} />
      <MenuList>
        <Grid container columnSpacing={25} rowSpacing={6}>
          {menuList}
        </Grid>
      </MenuList>
    </Paper>
  );
}

export default ExploreMenu;
