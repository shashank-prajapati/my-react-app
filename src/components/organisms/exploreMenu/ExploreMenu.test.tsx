import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import ExploreMenu from "./ExploreMenu";

describe("Testing Explore Menu", () => {
  it("should match the Explore Menu snapshot", () => {
    const { container } = render(<ExploreMenu />);
    expect(container).toMatchSnapshot();
  });

  it("should call on click method of button", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <ExploreMenu onClick={handleClick}></ExploreMenu>
    );
    fireEvent.click(screen.getByText("Entrepreneurship"));
    expect(handleClick).toHaveBeenCalled();
  });
});
