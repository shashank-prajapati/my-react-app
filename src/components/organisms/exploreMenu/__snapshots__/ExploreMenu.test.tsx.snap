// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Explore Menu should match the Explore Menu snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1o8kjj3-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-gg4vpm"
    >
      <span
        class="MuiTypography-root MuiTypography-subtitle3 css-cu8pq9-MuiTypography-root"
      >
        Explore by category
      </span>
      <span
        class="MuiTypography-root MuiTypography-subtitle3 css-1svm2tk-MuiTypography-root"
      >
        See recently added titles
      </span>
      <span
        class="MuiTypography-root MuiTypography-subtitle3 css-1svm2tk-MuiTypography-root"
      >
        See popular titles
      </span>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-ctpjxq-MuiDivider-root"
    />
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      role="menu"
      tabindex="-1"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-jno0xv-MuiGrid-root"
        tabindex="0"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Vector.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Entrepreneurship
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Politics.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Politics
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Marketing Sales.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Marketing & Sales
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Science.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Science
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Health Nutrition.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Health & Nutrition
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Personal Development.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Personal Development
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Economics.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Economics
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                History.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                History
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Communication Skills.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Communication Skills
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Corporate Culture.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                CorporateCulture
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Motivation Inspiration.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Motivation & Inspiration
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Money Investments.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Money & Investments
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Psychology.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Psychology
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Productivity.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Productivity
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                SexRelationship.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Sex & Relationship
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Nature Environment.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Nature & Environment
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                CareerSuccess.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Career & Success
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div>
            <div
              class="MuiBox-root css-ccqql9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                Education.svg
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
              >
                Education
              </p>
            </div>
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
